vectortile/src/main/scala/geotrellis/vectortile/protobuf/ProtobufTile.scala

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2016 Azavea.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.vectortile.protobuf
+
+import geotrellis.vector._
+import geotrellis.vectortile.{Layer, VectorTile}
+import geotrellis.vectortile.protobuf.internal._
+import geotrellis.spark.SpatialKey
+import geotrellis.spark.tiling.LayoutDefinition
+import geotrellis.vectortile.protobuf.internal.{vector_tile => vt}
+import geotrellis.vectortile.protobuf.internal.vector_tile.Tile.GeomType.{POINT, LINESTRING, POLYGON}
+
+import scala.collection.mutable.ListBuffer
+
+// --- //
+
+/**
+  * A concrete representation of a VectorTile, as one decoded from Protobuf
+  * bytes. This is the original/default type for VectorTiles.
+  *
+  * {{{
+  * import geotrellis.vectortile.protobuf._
+  *
+  * val bytes: Array[Byte] = ...  // from some `.mvt` file
+  * val key: SpatialKey = ...  // preknown
+  * val layout: LayoutDefinition = ...  // preknown
+  *
+  * val tile: VectorTile = ProtobufTile.fromBytes(bytes, key, layout)
+  * }}}
+  *
+  * @constructor This is not meant to be called directly. See this class's
+  * companion object for the available helper methods.
+  */
+case class ProtobufTile(
+  layers: Map[String, ProtobufLayer]
+) extends VectorTile {
+  /** Encode this VectorTile back into a mid-level Protobuf object. */
+  def toProtobuf: vt.Tile =
+    vt.Tile(layers = layers.values.map(_.toProtobuf).toSeq)
+
+  /** Encode this VectorTile back into its original form of Protobuf bytes. */
+  def toBytes: Array[Byte] = toProtobuf.toByteArray
+}
+
+object ProtobufTile {
+  /** Create a ProtobufTile masked as its parent trait. */
+  def fromPBTile(
+    tile: vt.Tile,
+    key: SpatialKey,
+    layout: LayoutDefinition
```

Comment:
We actually just want to pass in the Extent. SpatialKey/LayoutDefinition are Sparky things, and vector tiles shouldn't depend on sparky things (but be used by sparky things in the layer versions of them)